src/PrestaShopBundle/Controller/Admin/Sell/Catalog/AttachmentController.php

```diff
@@ -161,9 +152,10 @@ public function editAction($attachmentId, Request $request)
             }
         } catch (Exception $e) {
             $this->addFlash('error', $this->getErrorMessageForException($e, $this->getErrorMessages($e)));
-            if ($e instanceof AttachmentNotFoundException || $e instanceof ErrorException) {
-                return $this->redirectToRoute('admin_attachments_index');
-            }
+        }
+
+        if (!isset($attachmentInformation) || !isset($attachmentForm)) {
```

Comment:
It looks like a good idea but I dont understand why you remove the previous statement ?
```
if ($e instanceof AttachmentNotFoundException || $e instanceof ErrorException) {
...
```

Reply:
Phpstan wasnt passing this way. I figured, all we care is that those vars must be set and we dont need to check which exception fault is that. There were some controllers missing those catches, so i would need to dive into handlers and check which exception to catch. Seemed fast and convienient. Not sure if its good approach tho :D

Reply:
What I'm worried about is that some `ErrorException` is thrown from `$result = $attachmentFormHandler->handleFor((int) $attachmentId, $attachmentForm);` BUT `attachmentForm` and `attachmentInformation` are set

So please put back the exception 😉

Reply:
@RaimondasSapola, why you used this ErrorException, was it by mistake? I don't see it being thrown or used anywhere else.

Reply:
@zuk3975  If i remember correctly this exception can be thrown when deleting old file that doesn't exist. In legacy code when unlinking file warnings were suppressed and with Attachment Forms PR a possibility to throw an exception was introduced. So what was done is i just removed warning suppression and symfony handles it and throws an ErrorException.

Reply:
According to https://github.com/PrestaShop/PrestaShop/pull/15003#discussion_r323268073, I've came up with solution. Catch ErrorException as fast as possilbe and throw CannotUnlinkAttachmentException, by which I know in controller for sure why and when it happened, so i can add warning flash instead of error (because the edit action actually happens successful). @matks wdyt?

Reply:
@zuk3975 I love this solution 👍 !

Reply:
so its done :smile: